src/geometry/manifold/vector_bundle/tangent.lean

```diff
@@ -141,12 +141,12 @@ does not pick wrong instances. In this section, we record the right instances fo
 them, noting in particular that the tangent bundle is a smooth manifold. -/
 
 section
-local attribute [reducible] tangent_space
 
 variables {M} (x : M)
 
-instance : module 𝕜 (tangent_space I x) := by apply_instance
 instance : inhabited (tangent_space I x) := ⟨0⟩
+instance : normed_add_comm_group (tangent_space I x) := by delta_instance tangent_space
+instance : normed_space 𝕜 (tangent_space I x) := by delta_instance tangent_space
```

Comment:
These two look wrong: we will want to put a normed space structure on the tangent space in Riemannian manifolds, but in general it will have nothing to do with the norm coming from the unfolding on the (noncanonical) definition of the tangent space. What is canonical is only the topological vector space structure on `tangent_space I x`.

Reply:
I will remove these instances, nothing in this series of PRs actually depends on it.
I was wondering if these instances were fine, since it is indeed a non-canonical choice. 

However, if we actually want to put a different instance on `tangent_space` than on `E`, we have to be careful, since I'm cheating in various places, and actually abuse the definitional equality of `tangent_space I x` and `E` . This can be problematic. In the sphere eversion project, at the start we didn't have these instances, but then at some point I noticed that the `tangent_space I x` was treated by Lean as a normed space, using the instance on `E`. We were using some operations defined only on normed spaces, and we also wanted to use them on the tangent space (I'm not sure if the norm was actually important, or only superficially added as a requirement). 

The reason that we abuse the definitional equality of of `tangent_space I x` and `E` is because it's *really* annoying that `tangent_space I` is a dependent type. For example, if I have `f : tangent_space I x -> A` and `p : x = y` and `v : tangent_space y`, it is a lot easier to just write `f v` than to add a casting function. Furthermore, `simp` and `rw` will often refuse to rewrite two expressions of type `tangent_space I x` (because things would be type-incorrect), until you unfold `tangent_space`. And things like identifying `tangent_space (I.prod J) (M \x N)` with `tangent_space I M \x tangent_space J N` is only possible when unfolding `tangent_space`, but it would be really inconvenient to have to transport along the equivalence of these two vector spaces...

So I removed these non-canonical instances, but we probably have to do something different before adding instances on `tangent_space` incompatible with those on `E`.

Reply:
Yes, dependent types are a major pain, and abusing defeq is definitely meant to be possible here to avoid this pain, but we will have to be very careful with those instances as you say.